File: Stronghold/Stronghold.Application/DTOs/Request/UpdateMembershipPackageRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Stronghold.Application.DTOs.Request
{
    public class UpdateMembershipPackageRequest
    {
        [StringLength(50, MinimumLength = 2, ErrorMessage = "Naziv paketa mora imati između 2 i 50 karaktera.")]
        public string? PackageName { get; set; }

        [Range(0.01, 10000, ErrorMessage = "Cijena mora biti veća od 0 i manja ili jednaka 10000.")]
        public decimal? PackagePrice { get; set; }

        [StringLength(500, ErrorMessage = "Opis paketa može imati najviše 500 karaktera.")]
        public string? Description { get; set; }

        public bool? IsActive { get; set; }
    }
}

File: packages/stronghold_core/lib/models/requests/update_membership_package_request.dart
/// Matches backend UpdateMembershipPackageRequest exactly
class UpdateMembershipPackageRequest {
  final String? packageName;
  final double? packagePrice;
  final String? description;

  const UpdateMembershipPackageRequest({
    this.packageName,
    this.packagePrice,
    this.description,
  });

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    if (packageName != null) map['packageName'] = packageName;
    if (packagePrice != null) map['packagePrice'] = packagePrice;
    if (description != null) map['description'] = description;
    return map;
  }
}
